### PR TITLE
formulae*: use shared_library extension

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
First experiment on a few formula to see if the new DSL works well.
I will migrate as much as possible later on, and add a cop to enforce the usage of the new DSL.

File: Formula/avian.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ class Avian < Formula
   def install
     system "make", "use-clang=true"
     bin.install Dir["build/macosx-*/avian*"]
-    lib.install Dir["build/macosx-*/*.dylib", "build/macosx-*/*.a"]
+    lib.install Dir["build/macosx-*/#{shared_library("*")}", "build/macosx-*/*.a"]
```
Comment (SMillerDev) [inline Formula/avian.rb:26]: Wouldn't this be in a different directory?

Comment (iMichka) [reply Formula/avian.rb:26]: I see, you mean in a Linux directory. This is up to discussion. The question will be how I'll teach rubocop to ignore theses cases and not try to enforce shared_library. The cop will basically look at all the strings assigned to variables or arguments passed to functions and ask you to no hardcode .dylib or .so.

Comment (SMillerDev) [reply Formula/avian.rb:26]: I think you should just use something like an osname variable in the directory name here. Disclaimer: I don't know if we have one of those.

**Pull request closed without merging**